### PR TITLE
#6938 Disallow unknown keys in feature-flags configmap

### PR DESCRIPTION
**Pull request by minhoryang**
# Changes

- #6938 Disallow unknown keys in feature-flags configmap
- Add knownFeatureFlagKeys for handling unknown keys

/kind: bug

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- n/a Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- n/a Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Disallow unknown keys in feature-flags configmap
```

### CODE REVIEW

**Review by khrm — COMMENTED** [review]
/ok-to-test

File: pkg/apis/config/feature_flags.go [hunk 1]
```diff
@@ -148,13 +158,15 @@ func GetFeatureFlagsConfigName() string {
 
 // NewFeatureFlagsFromMap returns a Config given a map corresponding to a ConfigMap
 func NewFeatureFlagsFromMap(cfgMap map[string]string) (*FeatureFlags, error) {
+	cfgMapKeys := sets.StringKeySet(cfgMap)
```
Comment (Yongxuanzhang) [inline pkg/apis/config/feature_flags.go:161]: can we store all known key to a set and check if there are unknown keys?

Comment (minhoryang) [reply pkg/apis/config/feature_flags.go:161]: @Yongxuanzhang Oh I got it. That way could be better for future-proof and readability (specially for `DeprecatedOrIgnorableFeatureFlags`)

**Comment by minhoryang** [comment]
- cherry-pick/rebased for `tide`bot
- `make test-unit test-e2e` done.

File: pkg/apis/config/feature_flags.go [hunk 2]
```diff
@@ -118,6 +119,35 @@ const (
...
+	enableTektonOCIBundles,
+	enableAPIFields,
+	sendCloudEventsForRuns,
+	enforceNonfalsifiability,
+	verificationNoMatchPolicy,
+	enableProvenanceInStatus,
+	resultExtractionMethod,
+	maxResultSize,
+	setSecurityContextKey,
+	coscheduleKey,
```
Comment (QuanZhang-William) [inline pkg/apis/config/feature_flags.go:141]: We recently added a few new feature flag: https://github.com/tektoncd/pipeline/blob/7069889e796b0b12e12a3d1095a5f1a3d98c2739/pkg/apis/config/feature_flags.go#L102C1-L112C32

Please add them to the set as well 😁

**Review by QuanZhang-William — COMMENTED** [review]
Thanks for the PR ❤️, left a small comment.

**Comment by QuanZhang-William** [comment]
> /cc @tektoncd/operator-maintainers this might break us when we keep some flags a little bit longer than `pipeline`.

Hi @vdemeester, I'm not sure if I fully understand what do you mean by "keeping some flag flag longer than `pipeline`"?

**Comment by Yongxuanzhang** [comment]
If we have concerns, it may not be necessary to proceed this PR/issue?

**Pull request closed without merging**